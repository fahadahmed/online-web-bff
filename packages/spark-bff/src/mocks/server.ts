import { setupServer } from 'msw/node';
import a2pCustomerHandlers from 'datasources/a2p/customer/mocks/handlers';
import accessHandlers from 'datasources/user/access/mocks/handlers';
import accountSummaryHandlers from 'datasources/customer/accountsService/mocks/handlers';
import addressHandlers from 'datasources/location/address/mocks/handlers';
import airpointsHandlers from 'datasources/airpoints/mocks/handlers';
import availableNumbers from 'datasources/shop/availableNumbers/mocks/handlers';
import balanceServiceHandlers from 'datasources/finance/balanceService/mocks/handlers';
import billPaymentHandlers from 'datasources/finance/paymentService/billPayment/mocks/handlers';
import cartServicesHandlers from 'datasources/shop/cart/mocks/handlers';
import checkoutServicesHandlers from 'datasources/shop/checkout/mocks/handlers';
import connectionDiagnosticsHandlers from 'datasources/line/assure/connectionDiagnostics/mocks/handlers';
import connectionPromiseHandler from 'datasources/line/assure/connectionPromise/mocks/handlers';
import contentServiceHandlers from 'datasources/utility/content/mocks/handlers';
import digitalBillHandlers from 'datasources/finance/digitalBill/mocks/handlers';
import identityServiceHandlers from 'datasources/user/identityService/mocks/handlers';
import lineConfiguarationHandlers from 'datasources/line/lineConfiguration/mocks/handlers';
import lineDetailServiceHandlers from 'datasources/line/lineDetails/mocks/handlers';
import lineDetailsHandlers from 'datasources/line/lineSummary/mocks/handlers';
import lineSpendHistoryHandlers from 'datasources/line/spend/mocks/handlers';
import lineSubscriptionsService from 'datasources/line/lineSubscriptionsService/mocks/handlers';
import lineUsageDetailsMocks from 'datasources/line/usage/usageDetailsService/mocks';
import lineUsageHistoryMocks from 'datasources/line/usage/usageHistoryService/mocks';
import lineUsageMocks from 'datasources/line/usage/mocks/handlers';
import lineUsageTransactionMocks from 'datasources/line/usage/usageTransactionsService/mocks';
import networkSettingsHandlers from 'datasources/technology/networkSettingsService/mocks/handlers';
import orderHandlers from 'datasources/shop/order/mocks/handlers';
import paymentServiceHandlers from 'datasources/finance/paymentService/mocks/handlers';
import pointOfAddressHandlers from 'datasources/location/pointOfInterest/mocks/handlers';
import pointOfOutageHandlers from 'datasources/location/pointOfInterest/outagesPointOfInterest/mocks/handlers';
import preferenceHandlers from 'datasources/user/preference/mocks/handlers';
import prepaidBalanceServiceHandlers from 'datasources/finance/prepaidBalanceService/mocks/handlers';
import productDetailServiceHandlers from 'datasources/product/details/mocks/handlers';
import productFiltersServiceHandlers from 'datasources/product/filters/mocks/handlers';
import productOffersHandlers from 'datasources/product/offers/mocks/handlers';
import productPromotionsMPDServiceHandlers from 'datasources/product/promotions/mpd/mocks/handlers';
import profileServiceHandlers from 'datasources/marketing/profileService/mocks/handlers';
import resourceHandlers from 'datasources/utility/resource/mocks/handlers';
import roamingRatesHandlers from 'datasources/product/roamingRates/mocks/handlers';
import subscriptionActivationHandlers from 'datasources/line/subscriptionActivationService/mocks/handlers';
import topupHandlers from 'datasources/finance/paymentService/topup/mocks/handlers';
import transactionsListHandlers from 'datasources/line/transactionsService/mocks/handlers';
import troubleshootHandlers from 'datasources/line/assure/connectionTroubleshooter/mocks/handlers';
import userInfoHandlers from 'datasources/user/userInfo/mocks/handlers';
import walletServiceHandlers from 'datasources/finance/walletService/mocks/handlers';

const handlers = [
  ...a2pCustomerHandlers,
  ...accessHandlers,
  ...accountSummaryHandlers,
  ...addressHandlers,
  ...availableNumbers,
  ...airpointsHandlers,
  ...balanceServiceHandlers,
  ...billPaymentHandlers,
  ...cartServicesHandlers,
  ...checkoutServicesHandlers,
  ...connectionDiagnosticsHandlers,
  ...connectionPromiseHandler,
  ...contentServiceHandlers,
  ...digitalBillHandlers,
  ...identityServiceHandlers,
  ...lineConfiguarationHandlers,
  ...lineDetailServiceHandlers,
  ...lineDetailsHandlers,
  ...lineSpendHistoryHandlers,
  ...lineSubscriptionsService,
  ...lineUsageDetailsMocks,
  ...lineUsageHistoryMocks,
  ...lineUsageMocks,
  ...lineUsageTransactionMocks,
  ...networkSettingsHandlers,
  ...orderHandlers,
  ...paymentServiceHandlers,
  ...pointOfAddressHandlers,
  ...pointOfOutageHandlers,
  ...preferenceHandlers,
  ...prepaidBalanceServiceHandlers,
  ...productDetailServiceHandlers,
  ...productFiltersServiceHandlers,
  ...productOffersHandlers,
  ...productPromotionsMPDServiceHandlers,
  ...profileServiceHandlers,
  ...resourceHandlers,
  ...roamingRatesHandlers,
  ...subscriptionActivationHandlers,
  ...topupHandlers,
  ...transactionsListHandlers,
  ...troubleshootHandlers,
  ...userInfoHandlers,
  ...walletServiceHandlers,
];

const server = setupServer(...handlers);
export default server;
