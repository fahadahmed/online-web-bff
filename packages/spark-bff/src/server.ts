import { ApolloServer } from 'apollo-server-fastify';
import {
  ApolloServerPluginLandingPageDisabled,
  ApolloServerPluginLandingPageGraphQLPlayground,
} from 'apollo-server-core';
import { logger } from 'apolloLogger';
import context from 'context';
import dataSources from 'datasources';
import { ApolloLoggerPlugin, ApplicationLogEvent } from 'logging';
import schema from 'schema';
import { formatResponse } from 'utils/formatResponse';
import { isDevelopmentEnvironment } from 'utils/isDevelopmentEnvironment';

process.on('uncaughtException', (err) => {
  logger.error(new ApplicationLogEvent({ message: err.message, context: {} }));
  throw err;
});

const apolloServer = new ApolloServer({
  schema,
  dataSources,
  formatResponse,
  logger,
  context,
  introspection: isDevelopmentEnvironment(),
  plugins: [
    new ApolloLoggerPlugin(
      `BFF Server Started. BUILD_ID: ${process.env.BUILD_BUILDID}`,
    ),
    isDevelopmentEnvironment()
      ? ApolloServerPluginLandingPageGraphQLPlayground()
      : ApolloServerPluginLandingPageDisabled(),
  ],
});

export default apolloServer;
