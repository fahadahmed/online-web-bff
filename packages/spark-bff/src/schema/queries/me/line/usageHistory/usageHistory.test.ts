import { createTestClient, gql } from 'test/apolloTestUtils';
import { server } from 'test/server';
import usageHistoryMocks from 'datasources/line/usage/usageHistoryService/mocks';
import { NexusGenEnums } from 'generated/nexusTypes';

beforeEach(() => {
  server.use(...usageHistoryMocks);
});

function constructUsageHistoryQuery(
  usageType: NexusGenEnums['LineUsageHistoryUsageType'],
  interval: NexusGenEnums['LineHistoryInterval'],
) {
  return {
    query: gql`
      query UsageHistoryQuery(
        $usageType: LineUsageHistoryUsageType!
        $interval: LineHistoryInterval!
      ) {
        me {
          line(lineNumber: "033227715") {
            usageHistory(usageType: $usageType, interval: $interval) {
              accountType
              sharerType
              interval
              startDateTime
              endDateTime
              averagePeriodUsage {
                unit
                value
              }
              averageGroupPeriodUsage {
                unit
                value
              }
              summarisedPeriods {
                startDateTime
                endDateTime
                unbilled
                periodUsage {
                  unit
                  value
                }
                periodGroupUsage {
                  unit
                  value
                }
                periodBreakdown {
                  breakdownType
                  periodBreakdownId
                  periodUsage {
                    value
                    unit
                  }
                  periodGroupUsage {
                    unit
                    value
                  }
                  periodLimit {
                    value
                    unit
                  }
                  periodGroupLimit {
                    unit
                    value
                  }
                  periodRemaining {
                    value
                    unit
                  }
                  periodGroupRemaining {
                    unit
                    value
                  }
                }
              }
            }
          }
        }
      }
    `,
    variables: {
      usageType,
      interval,
    },
  };
}

test('hourly data usage', async () => {
  const { query } = createTestClient();
  const res = await query(constructUsageHistoryQuery('DATA', 'HOURLY'));

  expect(res.errors).toBeUndefined();
  expect(res.data!.me.line).toMatchSnapshot();
});

test('daily data usage', async () => {
  const { query } = createTestClient();
  const res = await query(constructUsageHistoryQuery('DATA', 'DAILY'));

  expect(res.errors).toBeUndefined();
  expect(res.data!.me.line).toMatchSnapshot();
});

test('monthly data usage', async () => {
  const { query } = createTestClient();
  const res = await query(constructUsageHistoryQuery('DATA', 'MONTHLY'));

  expect(res.errors).toBeUndefined();
  expect(res.data!.me.line).toMatchSnapshot();
});

test('monthly sms usage', async () => {
  const { query } = createTestClient();
  const res = await query(constructUsageHistoryQuery('SMS', 'MONTHLY'));

  expect(res.errors).toBeUndefined();
  expect(res.data!.me.line).toMatchSnapshot();
});

test('monthly voice usage', async () => {
  const { query } = createTestClient();
  const res = await query(constructUsageHistoryQuery('VOICE', 'MONTHLY'));

  expect(res.errors).toBeUndefined();
  expect(res.data!.me.line).toMatchSnapshot();
});
