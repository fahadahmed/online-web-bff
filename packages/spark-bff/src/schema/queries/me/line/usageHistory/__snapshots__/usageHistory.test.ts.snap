// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`daily data usage 1`] = `
Object {
  "usageHistory": Object {
    "accountType": "POSTPAID",
    "averageGroupPeriodUsage": null,
    "averagePeriodUsage": Object {
      "unit": "MB",
      "value": 768.00293,
    },
    "endDateTime": "2021-10-15T23:59:59+13:00",
    "interval": "DAILY",
    "sharerType": "NOT_SHARED",
    "startDateTime": "2021-10-07T11:00:00Z",
    "summarisedPeriods": Array [
      Object {
        "endDateTime": "2021-10-15T10:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 1500,
        },
        "startDateTime": "2021-10-14T11:00:00Z",
        "unbilled": true,
      },
      Object {
        "endDateTime": "2021-10-14T10:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-13T11:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-13T10:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-12T11:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-12T10:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-11T11:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-11T10:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 2048.007813,
        },
        "startDateTime": "2021-10-10T11:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T10:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 1024.003906,
        },
        "startDateTime": "2021-10-09T11:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-09T10:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 1024.003906,
        },
        "startDateTime": "2021-10-08T11:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-08T10:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 2048.007813,
        },
        "startDateTime": "2021-10-07T11:00:00Z",
        "unbilled": false,
      },
    ],
  },
}
`;

exports[`hourly data usage 1`] = `
Object {
  "usageHistory": Object {
    "accountType": "POSTPAID",
    "averageGroupPeriodUsage": null,
    "averagePeriodUsage": Object {
      "unit": "MB",
      "value": 85.333659,
    },
    "endDateTime": "2021-10-10T23:59:59Z",
    "interval": "HOURLY",
    "sharerType": "NOT_SHARED",
    "startDateTime": "2021-10-10T00:00:00Z",
    "summarisedPeriods": Array [
      Object {
        "endDateTime": "2021-10-10T23:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 567,
        },
        "startDateTime": "2021-10-10T23:00:00Z",
        "unbilled": true,
      },
      Object {
        "endDateTime": "2021-10-10T22:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 2048.007813,
        },
        "startDateTime": "2021-10-10T22:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T21:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T21:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T20:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T20:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T19:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T19:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T18:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T18:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T17:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T17:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T16:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T16:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T15:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T15:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T14:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T14:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T13:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T13:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T12:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T12:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T11:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T11:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T10:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T10:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T09:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T09:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T08:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T08:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T07:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T07:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T06:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T06:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T05:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T05:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T04:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T04:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T03:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T03:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T02:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T02:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T01:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T01:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-10-10T00:59:59Z",
        "periodBreakdown": null,
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MB",
          "value": 0,
        },
        "startDateTime": "2021-10-10T00:00:00Z",
        "unbilled": false,
      },
    ],
  },
}
`;

exports[`monthly data usage 1`] = `
Object {
  "usageHistory": Object {
    "accountType": "POSTPAID",
    "averageGroupPeriodUsage": Object {
      "unit": "MB",
      "value": 1899.448243,
    },
    "averagePeriodUsage": Object {
      "unit": "MB",
      "value": 1010.307618,
    },
    "endDateTime": "2021-11-12T10:59:59Z",
    "interval": "MONTHLY",
    "sharerType": "LEADER",
    "startDateTime": "2021-08-12T12:00:00Z",
    "summarisedPeriods": Array [
      Object {
        "endDateTime": "2021-11-12T10:59:59Z",
        "periodBreakdown": Array [
          Object {
            "breakdownType": "PLAN",
            "periodBreakdownId": "PLAN|2021-10-12T11:00:00.000Z|2021-11-12T10:59:59.000Z",
            "periodGroupLimit": Object {
              "unit": "MB",
              "value": 4096,
            },
            "periodGroupRemaining": Object {
              "unit": "MB",
              "value": 2431.703124,
            },
            "periodGroupUsage": Object {
              "unit": "MB",
              "value": 1664.296876,
            },
            "periodLimit": Object {
              "unit": "MB",
              "value": 4096,
            },
            "periodRemaining": Object {
              "unit": "MB",
              "value": 2912.757813,
            },
            "periodUsage": Object {
              "unit": "MB",
              "value": 1183.242187,
            },
          },
        ],
        "periodGroupUsage": Object {
          "unit": "MB",
          "value": 1664.296876,
        },
        "periodUsage": Object {
          "unit": "MB",
          "value": 1183.242187,
        },
        "startDateTime": "2021-10-12T11:00:00Z",
        "unbilled": true,
      },
      Object {
        "endDateTime": "2021-10-12T10:59:59Z",
        "periodBreakdown": Array [
          Object {
            "breakdownType": "PLAN",
            "periodBreakdownId": "PLAN|2021-09-12T12:00:00.000Z|2021-10-12T10:59:59.000Z",
            "periodGroupLimit": Object {
              "unit": "MB",
              "value": 4096,
            },
            "periodGroupRemaining": Object {
              "unit": "MB",
              "value": 3888.666016,
            },
            "periodGroupUsage": Object {
              "unit": "MB",
              "value": 207.333984,
            },
            "periodLimit": Object {
              "unit": "MB",
              "value": 4096,
            },
            "periodRemaining": Object {
              "unit": "MB",
              "value": 3987.5625,
            },
            "periodUsage": Object {
              "unit": "MB",
              "value": 108.4375,
            },
          },
        ],
        "periodGroupUsage": Object {
          "unit": "MB",
          "value": 207.333984,
        },
        "periodUsage": Object {
          "unit": "MB",
          "value": 108.4375,
        },
        "startDateTime": "2021-09-12T12:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-09-12T11:59:59Z",
        "periodBreakdown": Array [
          Object {
            "breakdownType": "PLAN",
            "periodBreakdownId": "PLAN|2021-08-12T12:00:00.000Z|2021-09-12T11:59:59.000Z",
            "periodGroupLimit": Object {
              "unit": "MB",
              "value": 4096,
            },
            "periodGroupRemaining": Object {
              "unit": "MB",
              "value": 504.437499,
            },
            "periodGroupUsage": Object {
              "unit": "MB",
              "value": 3591.562501,
            },
            "periodLimit": Object {
              "unit": "MB",
              "value": 4096,
            },
            "periodRemaining": Object {
              "unit": "MB",
              "value": 2183.822265,
            },
            "periodUsage": Object {
              "unit": "MB",
              "value": 1912.177735,
            },
          },
        ],
        "periodGroupUsage": Object {
          "unit": "MB",
          "value": 3591.562501,
        },
        "periodUsage": Object {
          "unit": "MB",
          "value": 1912.177735,
        },
        "startDateTime": "2021-08-12T12:00:00Z",
        "unbilled": false,
      },
    ],
  },
}
`;

exports[`monthly sms usage 1`] = `
Object {
  "usageHistory": Object {
    "accountType": "POSTPAID",
    "averageGroupPeriodUsage": null,
    "averagePeriodUsage": Object {
      "unit": "TEXT",
      "value": 238,
    },
    "endDateTime": "2021-11-08T10:59:59Z",
    "interval": "MONTHLY",
    "sharerType": "NOT_SHARED",
    "startDateTime": "2021-08-08T12:00:00Z",
    "summarisedPeriods": Array [
      Object {
        "endDateTime": "2021-11-08T10:59:59Z",
        "periodBreakdown": Array [
          Object {
            "breakdownType": "ADDITIONAL_INCLUDED",
            "periodBreakdownId": "ADDITIONAL_INCLUDED|2021-10-08T11:00:00.000Z|2021-11-08T10:59:59.000Z",
            "periodGroupLimit": null,
            "periodGroupRemaining": null,
            "periodGroupUsage": null,
            "periodLimit": null,
            "periodRemaining": null,
            "periodUsage": Object {
              "unit": "TEXT",
              "value": 203,
            },
          },
        ],
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "TEXT",
          "value": 203,
        },
        "startDateTime": "2021-10-08T11:00:00Z",
        "unbilled": true,
      },
      Object {
        "endDateTime": "2021-10-08T10:59:59Z",
        "periodBreakdown": Array [
          Object {
            "breakdownType": "PLAN",
            "periodBreakdownId": "PLAN|2021-09-08T12:00:00.000Z|2021-10-08T10:59:59.000Z",
            "periodGroupLimit": null,
            "periodGroupRemaining": null,
            "periodGroupUsage": null,
            "periodLimit": null,
            "periodRemaining": null,
            "periodUsage": Object {
              "unit": "TEXT",
              "value": 360,
            },
          },
        ],
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "TEXT",
          "value": 360,
        },
        "startDateTime": "2021-09-08T12:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-09-08T11:59:59Z",
        "periodBreakdown": Array [
          Object {
            "breakdownType": "PLAN",
            "periodBreakdownId": "PLAN|2021-08-08T12:00:00.000Z|2021-09-08T11:59:59.000Z",
            "periodGroupLimit": null,
            "periodGroupRemaining": null,
            "periodGroupUsage": null,
            "periodLimit": null,
            "periodRemaining": null,
            "periodUsage": Object {
              "unit": "TEXT",
              "value": 151,
            },
          },
        ],
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "TEXT",
          "value": 151,
        },
        "startDateTime": "2021-08-08T12:00:00Z",
        "unbilled": false,
      },
    ],
  },
}
`;

exports[`monthly voice usage 1`] = `
Object {
  "usageHistory": Object {
    "accountType": "POSTPAID",
    "averageGroupPeriodUsage": null,
    "averagePeriodUsage": Object {
      "unit": "MIN",
      "value": 171,
    },
    "endDateTime": "2021-11-08T10:59:59Z",
    "interval": "MONTHLY",
    "sharerType": "NOT_SHARED",
    "startDateTime": "2021-09-08T12:00:00Z",
    "summarisedPeriods": Array [
      Object {
        "endDateTime": "2021-11-08T10:59:59Z",
        "periodBreakdown": Array [
          Object {
            "breakdownType": "PLAN",
            "periodBreakdownId": "PLAN|2021-10-08T11:00:00.000Z|2021-11-08T10:59:59.000Z",
            "periodGroupLimit": null,
            "periodGroupRemaining": null,
            "periodGroupUsage": null,
            "periodLimit": null,
            "periodRemaining": null,
            "periodUsage": Object {
              "unit": "MIN",
              "value": 104,
            },
          },
        ],
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MIN",
          "value": 104,
        },
        "startDateTime": "2021-10-08T11:00:00Z",
        "unbilled": true,
      },
      Object {
        "endDateTime": "2021-10-08T10:59:59Z",
        "periodBreakdown": Array [
          Object {
            "breakdownType": "PLAN",
            "periodBreakdownId": "PLAN|2021-09-08T12:00:00.000Z|2021-10-08T10:59:59.000Z",
            "periodGroupLimit": null,
            "periodGroupRemaining": null,
            "periodGroupUsage": null,
            "periodLimit": null,
            "periodRemaining": null,
            "periodUsage": Object {
              "unit": "MIN",
              "value": 238,
            },
          },
        ],
        "periodGroupUsage": null,
        "periodUsage": Object {
          "unit": "MIN",
          "value": 238,
        },
        "startDateTime": "2021-09-08T12:00:00Z",
        "unbilled": false,
      },
    ],
  },
}
`;
